ide/lsp.client/src/org/netbeans/modules/lsp/client/LSPBindings.java

```diff
@@ -432,14 +447,19 @@ public void runOnBackground(Runnable r) {
     }
 
     private static void scheduleBackgroundTask(RequestProcessor.Task req) {
-        WORKER.post(req, DELAY);
+        synchronized(WORKER_SCHEDULED) {
+            if(! WORKER_SCHEDULED.containsKey(req)) {
+                WORKER_SCHEDULED.put(req, null);
+                WORKER.post(req, DELAY);
```

Comment:
Hmmm, I apparently forgot how this works - I somehow managed to think `post` will defer the task when the same task is posted again, but that is wrong. But I don't think there is a need to keep track of the tasks manually - the correct idiom is (I hope I am right this time!) `req.schedule(DELAY);`. Should take care of the re-scheduling/moving to a new time. Should work similarly to the map (I hope!) but much simpler.

Sorry for the trouble.

Reply:
I think the behavior of my proposed solution and the your suggestion is slightly different. 

From my observation (and my reading of the documentation of RequestProcessor), with RequestProcessor.Task#schedule, if an event occurs inside the 500ms timeframe, the execution of the background update will be postponed again for 500ms, if the the next event happens inside this 500ms frame, updates will be postponed again and so on. So the view is only updated, once the user stops changing things.

My solution would update after 500ms and again 500ms later.

I observed the behavior of the java editor and at least the navigator windows follows the principle described for RequestProcessor.Task#schedule and I never had a problem with that. 500ms seem to be short enough to be a good balance between wait time and resource usage.

I updated this PR accordingly.